src/mysql/connection.cr

```diff
@@ -28,8 +28,8 @@ class MySql::Connection < DB::Connection
       read_ok_or_err do |packet, status|
         raise "packet #{status} not implemented"
       end
-    rescue Errno
-      raise DB::ConnectionRefused.new
+    rescue ex : Errno
+      raise DB::ConnectionRefused.new(ex.message)
```

Comment:
How about setting the exception as the cause: `DB::ConnectionRefused.new(cause: ex)`. This is what the `PG` driver does.

Reply:
While I agree that is a nicer way in theory, it'd make the error message equal to what it is now, which would defeat the point behind this PR. Using the sample code as given in the Crystal GitBook and running it on a machine that doesn't have MySQL running, the current message is " (DB::ConnectionRefused)", while my PR changes it to "Error connecting to 'localhost:3306': Connection refused (DB::ConnectionRefused)", which in my opinion is clearer than the current message.

Reply:
In the next version of crystal it will print out the cause of the exception, so it depends on if the next crystal-db release will be before or after 0.25.0

Reply:
Is `DB::ConnectionRefused.new(cause: ex, message: ex.message)` an acceptable variant then? This does print the nice error message on 0.24.2

Reply:
Yeah I guess that's fine for now. After 0.25.0 is released, we should probably refactor `DB::ConnectionRefused` to have a single constructor which takes the cause, and the exception message should *always* be "Connection to the database was refused".

Reply:
I've made the requested change. With respect to 0.25.0, I think it is better to give as much detail as possible, even returning the name of the database that is being connected to can be helpful to expose obvious configuration errors.